docs/paper/dev/api/command-api/arguments/registry.mdx

```diff
@@ -0,0 +1,269 @@
+Finally, we enchant the item of the executing player's hand with whatever enchantment the sender chose at level 10 and send a success message.
+
+Here is how it looks in-game:
+<FullWidthVideo src={EnchantsRegistry}/>
+
+:::warning
+
+There are certain edge-cases, where this argument, due to missing registries on the client, will cause a **Network Protocol Error**.
+Basically, the only argument where this is the case right now is with the `STRUCTURE` registry key.
+
+```java
+// Registering this command will cause clients to not be able to connect to the server.
+final LiteralCommandNode<CommandSourceStack> invalidRegistryArgument = Commands.literal("registry-structure")
+    .then(Commands.argument("value", ArgumentTypes.resource(RegistryKey.STRUCTURE)))
+    .build();
+```
+
+Due to this fact, it is advised to only use the `STRUCTURE` registry key argument with a `resourceKey(...)` argument type and parse the values yourself.
+
+:::
+
+## Resource key argument
+For the client, there is barely any difference between the using `ArgumentTypes.resource` or `ArgumentTypes.resourceKey`. The only difference is that
+using `ArgumentTypes.resourceKey` does not provide **error checking**. We can visualize this using `RegistryKey.ITEM`.
+
+Here is the tab completion when using `ArgumentTypes.resource(RegistryKey.ITEM)`:
+<FullWidthVideo src={ResourceItem}/><br/>
+
+And here is the tab completion when using `ArgumentTypes.resourceKey(RegistryKey.ITEM)`:
+<FullWidthVideo src={ResourceKeyItem}/><br/>
+
+The resource argument provides a much cleaner user experience, whilst the `resourceKey` argument has one very important usecase: You get the raw
+`TypedKey<T>` returned as an argument result. This object is particularly useful, as it provides all information required to be able to retrieve
+a value from a registry yourself.
+
+:::tip
+
+Unless you have a specific reason for using the `resourceKey` argument over the `resource` one, the `resource` argument is preferred due to the client-side error
+checking and simple usability.
+
+:::
+
+### Direct code comparison
+Here is a simple code snipped on how one could use the `RegistryKey.ITEM` registry with a `resource` argument type:
+
+```java title="Item Resource"
+Commands.argument("item", ArgumentTypes.resource(RegistryKey.ITEM))
+    .executes(ctx -> {
+        final ItemType item = ctx.getArgument("item", ItemType.class);
+
+        if (ctx.getSource().getExecutor() instanceof Player player) {
+            player.getInventory().addItem(item.createItemStack());
+        }
+
+        return Command.SINGLE_SUCCESS;
+    });
+```
+
+Here is the same code, using a `resourceKey` argument type:
+
+```java title="Item Resource"
+Commands.argument("item", ArgumentTypes.resourceKey(RegistryKey.ITEM))
+    .executes(ctx -> {
+        final TypedKey<ItemType> itemKey = ctx.getArgument("value", TypedKey.class);
```

Comment:
This is the wrong argument type name and you should use RegistryArgumentExtractor instead